apps/lib/opt.c

```diff
@@ -565,19 +566,26 @@ int opt_intmax(const char *value, intmax_t *result)
     m = strtoimax(value, &endp, 0);
     if (*endp
             || endp == value
-            || ((m == INTMAX_MAX || m == INTMAX_MIN) && errno == ERANGE)
+            || ((m == INTMAX_MAX || m == INTMAX_MIN)
```

Comment:
Should really zero `errno` to avoid false positives.  It's only set on failure, not cleared on success.

Reply:
There's nothing saying that errno needs to be zeroed or anything like that on success.  *errno itself isn't an indication of if there was an error or not*, which is quite clear when reading errno(3):

> **errno**
> The  value  in  _errno_  is significant only when the return value of the
> call indicated an error (i.e., -1 from most system calls;  -1  or  NULL
> from  most  library  functions); ...

Reply:
If I want to parse the perfectly legitimate value INTMAX_MAX and any previous call to any library or system call returned with errno set to ERANGE, this code will flag an error.  That's not correct behaviour.

The only way to correctly check this is:
```c
errno = 0;
m = strtoimax(...);
if ((m == INTMAX_MAX || m == INTMAX_MIN) && errno = ERANGE)
```

Reply:
But...  we already do exactly that!

Reply:
Crap, missed the errno=0 :(
Ignore my ravings.